Frontend/src/Components/Notifications/Notifications.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from "react";
+import { useSocket } from "../Contexts/SocketContext";
+import Loading from "../Loading/Loading.jsx";
+import "./Notifications.css";
+
+const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
+
+const Notifications = () => {
+  const { socket } = useSocket();
+  const [notifications, setNotifications] = useState([]);
+  const [isLoading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchNotifications = async () => {
+      setLoading(true);
+      setError(null);
+      try {
+        const response = await fetch(`${API_KEY}/notifications`, {
+          method: "GET",
+          credentials: "include",
+          headers: {
+            Authorization: `Bearer ${localStorage.getItem("token")}`,
+          },
+        });
+        if (!response.ok) {
+          throw new Error("Failed to fetch notifications");
+        }
+        const data = await response.json();
+        setNotifications(data);
+      } catch (err) {
+        setError(err.message);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchNotifications();
+
+    socket.on("notification", (notification) => {
```

Comment:
does this mean notifications will only update when a user comes online (opens app.jsx) or when they open this notifications page?

Reply:
The notifications will update in real-time as long as the app is open and the socket connection is active, regardless of which page the user is on. This is because the SocketProvider from my socket context is at the top level of your app. it wraps all of the components. I also did a test in my backend and once i open my website, i get the console log that a user has connected.